File: Java/Easy/Accepted/sum-of-all-odd-length-subarrays_6.java
class Solution {
    public int sumOddLengthSubarrays(int[] arr) {
        
        int n = arr.length;
        int sum = 0;
        int[] prefix = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            prefix[i] = prefix[i - 1] + arr[i - 1];
        }
        for (int i = 0; i < n; i++) {
            for (int j = i; j < n; j += 2) {
                sum += prefix[j + 1] - prefix[i];
            }
        }
        return sum;
    }
}

File: Rust/Easy/WrongAnswer/sum-of-all-odd-length-subarrays_6.rs
impl Solution {
    pub fn sum_odd_length_subarrays(arr: Vec<i32>) -> i32 {
        
        let mut sum = 0;
        for i in 0..arr.len() {
            let mut j = 0;
            while j + i < arr.len() {
                sum += arr[i + j];
                j += 2;
            }
        }
        sum
    }
}